railties/lib/rails/generators/app_base.rb

```diff
@@ -280,6 +280,7 @@ def gem_for_database
         case options[:database]
         when "mysql"          then ["mysql2", [">= 0.3.18", "< 0.5"]]
         when "postgresql"     then ["pg", ["~> 0.18"]]
+        when "sqlserver"      then ["tiny_tds", nil]
```

Comment:
Because the `tiny_tds` gem is equal to `pg`. It is the low level `raw_connection`.